server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java

```diff
@@ -152,6 +154,26 @@ public void closeNoBlock() {
         delegate.closeNoBlock();
     }
 
+    /**
+     * This method returns a remote cluster alias for the given transport connection if it targets a node in the remote cluster.
+     * This method will return an optional empty in case the connection targets the local node or the node in the local cluster.
+     *
+     * @param connection the transport connection for which to resolve a remote cluster alias
+     * @return a cluster alias if the connection target a node in the remote cluster, otherwise an empty result
+     */
+    public static Optional<String> resolveRemoteClusterAlias(Transport.Connection connection) {
```

Comment:
Very optional (no pun intended), but I'm wondering if this is slightly better placed on `RemoteClusterService`. It's a little confusing that `RemoteConnectionManager` has a `clusterAlias` field (which is connection-specific) but this static method returns a different cluster alias instead. While that's the correct behavior, moving this method to RemoteClusterService would clarify that the cluster alias returned by this method has nothing to do with the clusterAlias stored on the connection manager. Again, this is for sure optional.

Reply:
Okay, I see your point. I don't feel strongly about keeping the resolve method in `RemoteConnectionManager`. Main reason I placed it here was to keep the creation and resolution of remote connections in proximity of each other and to hide the implementation details by keeping the `InternalRemoteConnection` private.